CannibalisticZombies/Assets/Scripts/Inventory.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class Inventory 
+{
+    private List<InventorySlot> inventory;
+    private int maxWeight;
+
```

Comment:
Lets add a UnityEvent here for when the weight gets updated. 
`public UnityEvent onWeightUpdated`

Make sure to initialize it in the constructor. We can invoke it using `eventName.Invoke()` in `UpdateCurrentWeight()` so that when we implement UI it will be directly tied to the inventory.